git_theta/checkpoints/base.py

```diff
@@ -65,6 +65,41 @@ def flatten(self):
     def unflatten(self):
         return utils.unflatten(self)
 
+    @classmethod
+    def diff(cls, m1: "Checkpoint", m2: "Checkpoint") -> "Checkpoint":
+        """Compute the diff between two checkpoints.
+
+        Parameters
+        ----------
+        m1 : Checkpoint
+            The new checkpoint
+        m2 : Checkpoint
+            The old checkpoint
+
+        Returns
+        -------
+        added : Checkpoint
+            Checkpoint containing the parameter groups added to m1
+        removed : Checkpoint
+            Checkpoint containing the parameter groups removed from m2
+        modified : Checkpoint
+            Checkpoint containing the parameter groups modified between m1 and m2
+        """
+        m1_flat = m1.flatten()
+        m2_flat = m2.flatten()
+        added = cls({k: v for k, v in m1_flat.items() if k not in m2_flat}).unflatten()
```

Comment:
It's disappointing python don't have a diff operation for dicts.

I was a bit surprised to see that this was a bit faster than set differences on the keys, can you add a comment so I don't try to come back and code golf it lolz?

When a has 100 keys and b has 102
```python
[ins] In [6]: %timeit {k: v for k, v in b.items() if k not in a}
5.58 µs ± 393 ns per loop (mean ± std. dev. of 7 runs, 100,000 loops each)

[ins] In [7]: %timeit {k: b[k] for k in b.keys() - a.keys()}
6.75 µs ± 1.4 µs per loop (mean ± std. dev. of 7 runs, 100,000 loops each)

[ins] In [8]: %timeit {k: b[k] for k in set(b.keys()) - set(a.keys())}
5.79 µs ± 148 ns per loop (mean ± std. dev. of 7 runs, 100,000 loops each)
```

When a has 100 keys and b has ~150 keys
```python
[ins] In [10]: %timeit {k: v for k, v in b.items() if k not in a}
8.42 µs ± 182 ns per loop (mean ± std. dev. of 7 runs, 100,000 loops each)

[ins] In [11]: %timeit {k: b[k] for k in set(b.keys()) - set(a.keys())}
10.8 µs ± 318 ns per loop (mean ± std. dev. of 7 runs, 100,000 loops each)
```